### ISSUE TITLE
Failed to instantiate class org.jenkinsci.plugins.ghprb.GhprbTrigger

### ISSUE THREAD

**Issue by bxu-mmd** [janinko/ghprb#351]
https://issues.jenkins-ci.org/browse/JENKINS-30684

Same as issue here in jenkins jira but wanted to add it here. i'm still experiencing this issue. it's weird. one of my configuration was setup and doesnt' experience this issue but the rest of my configs do.

when an item is saved as a ghprb the first time. it works fine. if we go into the config and try to save it again it throws the error: javax.servlet.ServletException: java.lang.RuntimeException: Failed to instantiate class org.jenkinsci.plugins.ghprb.GhprbTrigger from....

if we uncheck the gtprb, it'll save. And if we check it after we've saved it it'll work.

If it's already checked and saved, and we try to save the config again it throws the error.

**Comment by bxu-mmd** [comment]
downgrading jenkins to the latest stable version didn't help

**Comment by bxu-mmd** [comment]
javax.servlet.ServletException: java.lang.RuntimeException: Failed to instantiate class org.jenkinsci.plugins.ghprb.GhprbTrigger from {"gitHubAuthId":"0978d91e-fe0f-4275-938a-47951202f9e5","adminlist":"bxu-mmd mmd-devops","useGitHubHooks":true,"triggerPhrase":"","onlyTriggerPhrase":false,"autoCloseFailedPullRequests":false,"skipBuildPhrase":".*\\[skip\\W+ci\\].*","displayBuildErrorsOnDownstreamBuilds":false,"cron":"H/5 * * * *","whitelist":"","orgslist":"","allowMembersOfWhitelistedOrgsAsAdmin":false,"permitAll":true,"buildDescTemplate":"hey hey","whiteListTargetBranches":{"branch":""},"extensions":{"stapler-class":"","$class":""}}
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:796)

**Comment by bxu-mmd** [comment]
the issue seems to be that whenever we enter back into the configuration. it adds an empty trigger setup which we have to manually delete. else it throws the error shown above.

**Comment by bxu-mmd** [comment]
alright for anyone else having this issue. The issue is that the you must have an application setup that triggers a build commit status.  this can be configured in the jenkins configuration or job configuration. the values can be empty but this application trigger must be added.

**Comment by juharris** [comment]
I'm not sure what you mean by "triggers a build commit status" but I got around this in my job by unchecking "GitHub Pull Request Builder" under "Build Triggers".  Then I clicked "Apply" to save it.  Then I re-added the trigger by checking off the box or "GitHub Pull Request Builder".  I then deleted every element in the "Trigger Setup" section of "GitHub Pull Request Builder", even the empty element that had a delete option.
